File: RAI/Clase13/Python/Gato.py
from Animal import Animal

class Gato(Animal):
    def __init__(self, nombre: str, familia: str, especie: str, pelaje: str):
        super().__init__(nombre, familia, especie)
        self.pelaje = pelaje
    
    def ronrronea(self):
        print(self.nombre + " esta ronrroneando")
    
    #@override
    def hace_sonido(self):
        print(self.nombre + " hace Miauu...")

File: RAI/Clase13/Java/Gato.java
//package RAI.Clase13.Java;

//extends se refiere a la herencia a una superclase
public class Gato extends Animal{
    private String pelaje;

    public Gato(String nombre, String familia, String especie, String pelaje){
        //super() permite acceder al metodo constructor de la superclase
        super(nombre, familia, especie);
        this.pelaje = pelaje;
    }

    public void ronrronea(){
        System.out.println(this.nombre + " ronrronea...");
    }

    @Override
    public void haceSonido(){
        System.out.println(this.nombre + " hace miauu...");
    }

}
